src/Agent/NewRelic/Agent/Core/DataTransport/HttpCollectorWire.cs

```diff
@@ -46,76 +51,108 @@ private enum Direction
                 var uncompressedByteCount = bytes.Length;
                 var requestPayload = GetRequestPayload(bytes);
-                var request = BuildRequest(uri, connectionInfo, requestPayload);
 
-                // Check serializedData length < MaxPayloadSizeInBytes before sending
-                // if > silently drop the payload for now, ideally want to be able split the data
-                // into chunks to send
                 if (requestPayload.Data.Length > _configuration.CollectorMaxPayloadSizeInBytes)
                 {
-                    // Log that the payload is being dropped. 
+                    // Log that the payload is being dropped
                     Log.ErrorFormat("Request({0}): Dropped large payload: size: {1}, max_payload_size_bytes={2}", requestGuid, requestPayload.Data.Length, _configuration.CollectorMaxPayloadSizeInBytes);
                     _agentHealthReporter.ReportSupportabilityPayloadsDroppeDueToMaxPayloadSizeLimit(method);
                     return "{}"; // mimics what is sent by NoOpCollectorWire
                 }
 
-                var response = SendRequest(request, requestPayload.Data);
+                var httpClient = _httpClientFactory.CreateClient(connectionInfo.Proxy);
+
+                var request = new HttpRequestMessage
+                {
+                    RequestUri = uri
+                };
+
+
+                request.Headers.Add("User-Agent", $"NewRelic-DotNetAgent/{AgentInstallConfiguration.AgentVersion}");
+                request.Headers.Add("Timeout", _configuration.CollectorTimeout.ToString());
+
+                request.Headers.Add("Connection", "keep-alive");
+                request.Headers.Add("Keep-Alive", "true");
+                request.Headers.Add("ACCEPT-ENCODING", "gzip");
 
-                _agentHealthReporter.ReportSupportabilityDataUsage("Collector", method, uncompressedByteCount, new UTF8Encoding().GetBytes(response).Length);
+                var content = new ByteArrayContent(requestPayload.Data);
+                var encoding = (requestPayload.IsCompressed) ? requestPayload.CompressionType.ToLower() : "identity";
+                content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/octet-stream");
+                content.Headers.Add("Content-Encoding", encoding);
+                content.Headers.Add("Content-Length", requestPayload.Data.Length.ToString());
+
+                request.Content = content;
+
+                foreach (var header in _requestHeadersMap)
+                {
+                    request.Headers.Add(header.Key, header.Value);
+                }
+
+                if (_configuration.PutForDataSend)
+                {
+                    request.Method = HttpMethod.Put;
+                }
+                else
+                {
+                    request.Method = HttpMethod.Post;
+                }
+
+                var response = httpClient.SendAsync(request).GetAwaiter().GetResult();
```

Comment:
`response` should be inside a `using()` block to ensure it is correctly disposed.